lib/YuiRestClient.pm

```diff
@@ -17,110 +17,129 @@ use constant API_VERSION => 'v1';
 
 use testapi;
 use utils qw(enter_cmd_slow type_line_svirt save_svirt_pty zypper_call);
-use Utils::Backends qw(is_pvm is_ipmi);
+use Utils::Backends;
 use YuiRestClient::App;
 use YuiRestClient::Wait;
+use Utils::Architectures 'is_s390x';
 
-our $interval = 1;
-our $timeout  = 10;
-our $app;
+my $app;
+my $port;
+my $host;
 
-sub set_interval {
-    $interval = shift;
+sub get_app {
+    my (%args) = @_;
+    $app = init_app(%args) unless $app;
+    return $app;
 }
 
-sub set_timeout {
-    $timeout = shift;
+sub get_host {
+    return $host;
 }
 
-sub set_app {
-    $app = shift;
+sub get_port {
+    return $port;
 }
 
-sub get_app {
-    return $app;
+sub set_interval {
+    my ($interval) = @_;
+    $app->get_widget_controller()->set_interval($interval);
 }
 
-sub get_yui_params_string {
-    my $port = get_required_var('YUI_PORT');
-    return "YUI_HTTP_PORT=$port YUI_HTTP_REMOTE=1 YUI_REUSE_PORT=1";
+sub set_timeout {
+    my ($timeout) = @_;
+    $app->get_widget_controller()->set_timeout($timeout);
 }
 
-sub connect_to_app {
-    my $port = get_required_var('YUI_PORT');
-    my $host = get_required_var('YUI_SERVER');
-    die "Cannot set libyui REST API server" unless $host;
-    record_info('PORT',   "Used port for libyui: $port");
-    record_info('SERVER', "Connecting to: $host");
-    my $app = YuiRestClient::App->new({
+sub init_app {
+    my (%args)       = @_;
+    my $timeout      = $args{timeout}  || 10;
```

Comment:
we could group these default values using constant to make uniform like with API_VERSION so it is easy to read at the beginning of the file.

Reply:
Done.